dask_jobqueue/local.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import os
+import subprocess
+
+from .job import Job, JobQueueCluster, job_parameters, cluster_parameters
+
+logger = logging.getLogger(__name__)
+
+
+class LocalJob(Job):
+    __doc__ = """ Use Dask Jobqueue with local bash commands
+
+    This is mostly for testing.  It uses all the same machinery of
+    dask-jobqueue, but rather than submitting jobs to some external job
+    queueing system, it launches them locally.  For normal local use, please
+    see ``dask.distributed.LocalCluster``
+
+    Parameters
+    ----------
+    {job}
+    """.format(
+        job=job_parameters
+    )
+
+    config_name = "local"
+
+    def __init__(
+        self,
+        *args,
+        queue=None,
+        project=None,
+        resource_spec=None,
+        walltime=None,
+        job_extra=None,
+        config_name="local",
+        **kwargs
+    ):
+        # Instantiate args and parameters from parent abstract class
+        super().__init__(*args, config_name=config_name, shebang="", **kwargs)
+
+        # Declare class attribute that shall be overridden
+        header_lines = []
+        self.job_header = "\n".join(header_lines)
+
+        logger.debug("Job script: \n %s" % self.job_script())
+
+    def _submit_job(self, script_filename):
+        # Should we make this async friendly?
+        with open(script_filename) as f:
+            text = f.read().strip().split()
+        self.process = subprocess.Popen(
+            text, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        # TODO this should raise if self.process.returncode != 0. Refactor
+        # Job._call to be able to return process (so that we can return self.process.pid below)
```

Comment:
In the `LocalJob` case things are a little different.  Normally our command just calls `qsub` to submit a job, and then returns quickly.  In this case the process actually runs `dask-worker`, so it doesn't return until the worker goes away.

Reply:
I realised this afterwards. The error I had to  debug in the PBS case (https://github.com/dask/dask-jobqueue/pull/307#issuecomment-526649040 failure with LocalJob in the PBS case) was because basically starting `dask-worker` would fail with an error (the test is run as `pbsuser` and can does to create the `global.lock` or the `worker-bla` folder). It would be great if there was a way to fail early in this case.

A few alternatives I can see, not sure which one is the less fragile:
* read a few more lines from `self.process.stderr` and raise an exception if ERROR is in the stderr
* read a few more lines from `self.process.stderr` and raise an exception if "Start worker at" is not in the stderr
* wait for some time, say 1s, poll and if the process has finished raise an exception with the full stderr.